python/modem-firmware-1.3+/nrf_cloud_device_mgmt.py

```diff
@@ -0,0 +1,717 @@
+
+    if job_id:
+        print('Created job: {}'.format(job_id))
+
+    return job_id
+
+def print_device_list(device_list):
+    if len(device_list) == 0:
+        return
+
+    print("\nName,   ID,   App[Name, Version, FOTA support],   Modem[Version, FOTA support],   BOOT[Version, FOTA support],   Tags[]")
+    print("-----------------------------------------------------------------------------------------------------------------------")
+    for dev in device_list:
+        print(dev)
+    print("")
+
+def main():
+
+    if not len(sys.argv) > 1:
+        raise RuntimeError("No input provided")
+
+    args = parse_args()
+
+    # determine requested FOTA type
+    fota_type = None
+    for type in updateBundle.fotaType:
+        if type.name.casefold() == args.type.casefold():
+            fota_type = type
+            break
+
+    if fota_type is None:
+        raise RuntimeError('Invalid FOTA update type specified: \'{}\''.format(args.type))
+    elif fota_type is not updateBundle.fotaType.MODEM:
+        args.tag_list = True
+
+    # get update bundles of the requested FOTA type
+    print('Getting \'{}\' update bundles...'.format(fota_type.name))
+    bundles = get_requested_bundles(args.apikey, fota_type.name)
+
+    if len(bundles) == 0:
+        print('No \'{}\' bundles found'.format(fota_type.name))
+        return
+
+    print('Obtained {} \'{}\' update bundles'.format(len(bundles), fota_type.name))
+
+    update_by = updateBy.BASE_FW_VER
+
+    if args.dev_id:
+        print('Getting device {}...'.format(args.dev_id))
+        update_by = updateBy.DEV_ID
+    else:
+        print('Getting all devices...')
+        if args.tag_list or args.tag:
+            update_by = updateBy.TAG
+
+    devices = get_device_list(args.apikey, None, args.dev_id)
+    if len(devices) == 0:
+        print('No devices found')
+        return
+
+    # get a list of devices that support the requested FOTA type
+    requested_devices = []
+    if fota_type == updateBundle.fotaType.APP:
+        requested_devices = [d for d in devices if d.app_fota]
+    elif fota_type == updateBundle.fotaType.BOOT:
+        requested_devices = [d for d in devices if d.boot_fota]
+    elif fota_type == updateBundle.fotaType.MODEM:
+        requested_devices = [d for d in devices if d.mfw_fota]
+
+    # display devices if requested
+    if args.ad:
+        print_device_list(devices)
+    elif args.rd:
+        print_device_list(requested_devices)
+
+    print('{} of {} devices support \'{}\' FOTA updates'.format(len(requested_devices), len(devices), fota_type.name))
+
+    if len(requested_devices) == 0:
+        return
+
+    if fota_type == updateBundle.fotaType.MODEM:
+        handle_modem_updates(args.apikey, bundles, requested_devices, update_by,
+                             args.tag, args.bundle_id, args.name, args.desc, args.apply)
+    elif fota_type == updateBundle.fotaType.APP:
+        print('APP FOTA update creation not yet implemented')
```

Comment:
Please create a Jira ticket for this and BOOT.

Reply:
IRIS-4027
IRIS-4028